app/models/appointment.rb

```diff
@@ -65,6 +65,10 @@ def self.eligible_for_reminders(days_overdue: 3)
     overdue_by(days_overdue)
       .joins(:patient)
       .merge(Patient.contactable)
+      .includes(:communications)
+      .where(communications: {id: nil})
+      .includes(:appointment_reminders)
```

Comment:
I think you want to use an explicit [left_joins](https://apidock.com/rails/ActiveRecord/QueryMethods/left_outer_joins) here, as `includes` doesn't always necessarily do a left outer.